src/System.Private.CoreLib/shared/System/Type.Helpers.cs

```diff
@@ -271,55 +271,55 @@ public virtual MemberInfo[] FindMembers(MemberTypes memberType, BindingFlags bin
             {
                 for (i = 0; i < m.Length; i++)
                     if (m[i] != null)
-                        ret[cnt++] = m[i];
+                        ret[cnt++] = m[i]!;
             }
 
             // Copy the Constructors
             if (c != null)
             {
                 for (i = 0; i < c.Length; i++)
                     if (c[i] != null)
-                        ret[cnt++] = c[i];
+                        ret[cnt++] = c[i]!;
```

Comment:
For all of these cases where a `!` is needed on an indexer access even though we just checked, please add `// TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/34644` as a comment next to it.

Reply:
👌